lib/Runtime/Library/DateImplementation.h

```diff
@@ -418,7 +453,15 @@ namespace Js {
             bs->AppendChars(_u(' '));
 
             // year is directly after day, month, daydigit for IE11+
-            bs->AppendChars(pymd->year, 10, ConvertLongToString);
+            if ((pymd->year) > 0)
+            {
+                bs->AppendChars(pymd->year, 10, ConvertUInt32ToString_ZeroPad_4);
+            }
+            else
+            {
+                bs->AppendChars(_u('-'));
+                bs->AppendChars(-(pymd->year), 10, ConvertUInt32ToString_ZeroPad_4);
```

Comment:
Rather than passing in a negated negative number, maybe use a local variable named to make the intent more clear? Like
`positiveYear = -(pymd->year); // pymd->year is negative`